crates/parser/src/grammar/expressions.rs

```diff
@@ -120,16 +120,32 @@ pub(super) fn stmt(p: &mut Parser<'_>, semicolon: Semicolon) {
             // fn f() { let x: i32; }
             types::ascription(p);
         }
+
+        let mut is_block_like_expr_after_eq = false;
         if p.eat(T![=]) {
             // test let_stmt_init
             // fn f() { let x = 92; }
-            expressions::expr(p);
+            let expr = expressions::expr(p);
+
+            if let Some(expr) = expr {
+                is_block_like_expr_after_eq = match expr.kind() {
+                    IF_EXPR | WHILE_EXPR | FOR_EXPR | LOOP_EXPR | MATCH_EXPR | BLOCK_EXPR => true,
```

Comment:
Can we extract this match into an assocaited function on `BlockLike` since this is duplicated now and runs risk of going out of sync from the other place

Reply:
Yes